// textlint/lib/src/index.js
// LICENSE : MIT
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadFixerFormatter = exports.loadLinterFormatter = exports.loadTextlintrc = exports.createLinter = exports.TextLintCore = exports.TextFixEngine = exports.TextLintEngine = exports.textlint = exports.cli = void 0;
/**
 * Command line interface
 */
var cli_js_1 = require("./cli.js");
Object.defineProperty(exports, "cli", { enumerable: true, get: function () { return cli_js_1.cli; } });
/**
 * @deprecated use New APIs
 */
var textlint_js_1 = require("./textlint.js");
Object.defineProperty(exports, "textlint", { enumerable: true, get: function () { return textlint_js_1.textlint; } });
/**
 * TextLintEngine is a wrapper around `textlint` for linting **multiple** files
 * include formatter, detecting utils
 * You can see engine/textlint-engine-core.js for more detail
 * @deprecated use New APIs
 */
var textlint_engine_js_1 = require("./DEPRECATED/textlint-engine.js");
Object.defineProperty(exports, "TextLintEngine", { enumerable: true, get: function () { return textlint_engine_js_1.TextLintEngine; } });
/**
 * TextFixEngine is a wrapper around `textlint` for linting **multiple** files
 * include formatter, detecting utils
 * You can see engine/textlint-engine-core.js for more detail
 * @deprecated use New APIs
 */
var textfix_engine_js_1 = require("./DEPRECATED/textfix-engine.js");
Object.defineProperty(exports, "TextFixEngine", { enumerable: true, get: function () { return textfix_engine_js_1.TextFixEngine; } });
/**
 * Core API for linting a **single** text or file.
 * @deprecated use New APIs or @textlint/kernel
 */
var textlint_core_js_1 = require("./DEPRECATED/textlint-core.js");
Object.defineProperty(exports, "TextLintCore", { enumerable: true, get: function () { return textlint_core_js_1.TextLintCore; } });
/* = New APIs */
/**
 * @see {https://textlint.org/docs/use-as-modules.html#new-apis}
 * @example
 *
 * ```js
 *  import { createLinter, loadTextlintrc, loadLinterFormatter } from "textlint";
 *  const descriptor = await loadTextlintrc();
 *  const linter = createLinter({
 *      descriptor
 *  });
 *  const results = await linter.lintFiles(["*.md"]);
 *  const formatter = await loadLinterFormatter({ formatterName: "stylish" })
 *  const output = formatter.format(results);
 *  console.log(output);
 *  ```
 */
var createLinter_js_1 = require("./createLinter.js");
Object.defineProperty(exports, "createLinter", { enumerable: true, get: function () { return createLinter_js_1.createLinter; } });
var TextlintrcLoader_js_1 = require("./loader/TextlintrcLoader.js");
Object.defineProperty(exports, "loadTextlintrc", { enumerable: true, get: function () { return TextlintrcLoader_js_1.loadTextlintrc; } });
var formatter_js_1 = require("./formatter.js");
Object.defineProperty(exports, "loadLinterFormatter", { enumerable: true, get: function () { return formatter_js_1.loadLinterFormatter; } });
Object.defineProperty(exports, "loadFixerFormatter", { enumerable: true, get: function () { return formatter_js_1.loadFixerFormatter; } });
